packages/cw721/src/msg.rs

```diff
@@ -83,23 +123,35 @@ pub enum Cw721ExecuteMsg<TMetadataExtension, TMetadataExtensionMsg> {
 }
 
 #[cw_serde]
-pub struct Cw721InstantiateMsg {
+pub struct Cw721InstantiateMsg<TCollectionExtensionMsg> {
     /// Name of the NFT contract
     pub name: String,
     /// Symbol of the NFT contract
     pub symbol: String,
+    /// Optional extension of the collection metadata
+    pub collection_info_extension: TCollectionExtensionMsg,
 
     /// The minter is the only one who can create new NFTs.
     /// This is designed for a base NFT that is controlled by an external program
     /// or contract. You will likely replace this with custom logic in custom NFTs
     pub minter: Option<String>,
 
+    /// Sets the creator of collection. The creator is the only one eligible to update `CollectionInfo`.
+    pub creator: Option<String>,
```

Comment:
Optionally a `creator` can be passed in `Cw721InstantiateMsg`. If none - then collection info is immutable!